tide_core.install

```diff
@@ -1446,3 +1446,60 @@ function tide_core_update_8061() {
   $field_storage->setSetting('allowed_values', $allowed_values);
   $field_storage->save();
 }
+
+/**
+ * Update table of contents and display headings fields.
+ */
+function tide_core_update_8063() {
+  // Remove redundant existing listing key values from DB.
+  $display_headings_tables = [
+    'node__field_node_display_headings',
+    'node_revision__field_node_display_headings',
+  ];
+  foreach ($display_headings_tables as $table) {
+    $query = \Drupal::database();
+    if ($query->schema()->tableExists($table)) {
+      $query = $query->delete($table);
+      $orGroup = $query->orConditionGroup()
+        ->condition('field_node_display_headings_value', 'showH2AndAccordionTitle')
+        ->condition('field_node_display_headings_value', 'showH2AndH3AndAccordionTitle');
+      $query->condition($orGroup);
+      $query->execute();
+    }
+  }
+  // Rename values for existing keys and remove the redundant options.
+  $field_storage = FieldStorageConfig::loadByName('node', 'field_node_display_headings');
+  if (!empty($field_storage)) {
+    $allowed_values = [];
+    $allowed_values['showH2'] = t('Show level 2 (H2) only.');
+    $allowed_values['showH2AndH3'] = t('Show level 2 (H2) and level 3 (H3).');
+    $field_storage->setSetting('allowed_values', $allowed_values);
+    $field_storage->save();
+  }
+
+  // Update field_show_table_of_content and display headings.
+  $field_configs = [
+    'field_show_table_of_content' => [
+      'label' => t('Show table of contents'),
+      'description' => t("The table of contents displays a page's heading structure (i.e. H2 and H3), with anchor links to jump down to the content."),
+    ],
+    'field_node_display_headings' => [
+      'label' => t('Choose headings to display'),
+      'description' => '',
+    ],
+  ];
+  foreach ($field_configs as $field_config => $config) {
+    $field_storage = FieldStorageConfig::loadByName('node', $field_config);
+    if (!empty($field_storage)) {
+      $bundles = $field_storage->getBundles();
+      foreach ($bundles as $bundle) {
```

Comment:
Although the field config seats in every module separately, but to apply the same changes on every module that using this field. It makes more sense to put this in tide_core, which will apply the changes to the content module that are project specific as well.